### DISCUSSION TITLE
How can I change the color of svg in next/image?

### DISCUSSION THREAD

**Discussion by Dann1y** [vercel/next.js#20993] [Help] [answered]
```
import Image from 'next/image'

*(skip codes)*

<Image src="/emotion.svg" alt="emtion" width={50} height={50} />
```
I want to change svg color in next/image(Next.js).

But in css,
```
img {
  fill="#ffffff"
}
```
is not working.

How can I do ?

**Comment by yahya-aghdam** [comment, marked as answer]
First install this:
`npm install --save-dev babel-plugin-inline-react-svg`


Create a .babelrc file in your root directory if it's not exist and put this code in it:
```
{
  "presets": ["next/babel"],
  "plugins": ["inline-react-svg"]
}
```


import the svg file with a name you choose like component:
`import MyEmotion from '../emotion.svg'`


Then use it:
`<MyEmotion width={100} height={100} fill={'red'} />`

**Reply by Edweis** [reply]
Won't that skip SVG optimization by NextJs ?

**Reply by yahya-aghdam** [reply]
> Won't that skip SVG optimization by NextJs ?

By default, NextJs image optimization dose not apply on SVG

https://nextjs.org/docs/basic-features/image-optimization

**Reply by Xam-Mlr** [reply]
The expected fill property is a boolean in my case ...
Why isn't it working ? (I'm on typescript)

**Reply by ELATTAR-Ayoub** [reply]
I'm using Typescript and I have this error:
**Error:** Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: object.

**Reply by dancherb** [reply]
This no longer works in next.js if using @next/font:

`@next/font" requires SWC although Babel is being used due to a custom babel config being present.`

**Reply by kneza23** [reply]
anyone found a way around `next/font` error?

**Reply by 0xhohenheim** [reply]
[SVGR](https://react-svgr.com/docs/next/) + [this config](https://github.com/gregberge/svgr/issues/860#issuecomment-1531904399) worked for me but I don't really like it, all the other SVGs I configured with `next/image` stopped working.

Need a more appropriate solution.

**Reply by HAjiMa666** [reply]
I found, you solve it finally? 
can you tell me  how to solve it?

**Comment by allicanseenow** [comment]
Is there an answer without a 3rd party plugin (by using "next/image") if we optimize SVG with the new next.config option in nextjs 12?

**Comment by Alynva** [comment]
My workaround using the SVG as mask (mono colored):

```js
export default function SolidSvg({ width, height, path, color = "black", fit = false }) {
	return <div>
		<style jsx>{`
			div {
				width: ${width};
				height: ${height};
				background-color: ${color};
				mask: url(${path});
				mask-repeat: no-repeat;
				mask-position: center;
				mask-size: ${fit ? "contain": "auto"};
			}
		`}</style>
	</div>
}
```

The downside is that you must set the size of the element, it won't auto-resizes to the SVG size, but you can set the `fit` flag to resize the mask to the element size instead.

**Comment by foadbotan** [comment]
If the svg is simple enough you can place it directly in your page jsx, instead of in an image <Image />. Then you can add fill="currentColor" to the svg so it inherits the default text color, or set the color to whatever you want.

**Reply by Dekita** [reply]
mucho gracias <3

**Reply by SKLC1** [reply]
can you give an example?

**Reply by SKLC1** [reply]
i just figured it out if anyone needed, you need to export the svgs like so:
```javascript
// this is how you export colorable icons export (svg must have "fill="currentColor" OR stroke="currentColor"), cannot have other stroke and no other fills are allowed
export { default as ArrowNarrowUpRight } from "./arrows/arrow-narrow-up-right.svg";
```
